pkg/controllers/podgroup_controller_test.go

```diff
@@ -146,47 +147,49 @@ func Test_Run(t *testing.T) {
 			desiredGroupPhase: v1alpha1.PodGroupPending,
 		},
 		{
-			name:              "Group status convert from running to pending",
+			name:              "Group status keeps running",
 			pgName:            "pg10",
-			minMember:         2,
-			podNames:          []string{},
+			minMember:         4,
+			podNames:          []string{"pod101", "pod102"},
 			podPhase:          v1.PodPending,
 			previousPhase:     v1alpha1.PodGroupRunning,
-			desiredGroupPhase: v1alpha1.PodGroupPending,
+			desiredGroupPhase: v1alpha1.PodGroupRunning,
 		},
 	}
 	for _, c := range cases {
 		t.Run(c.name, func(t *testing.T) {
 			controller, kClient := setUp(ctx, c.podNames, c.pgName, c.podPhase, c.minMember, c.previousPhase, c.podGroupCreateTime, nil)
 			// 0 means not set
+			ps := makePods(c.podNames, c.pgName, c.podPhase, nil)
 			if len(c.podNextPhase) != 0 {
```

Comment:
This test code didn't work properly if test case didn't have podNextPhase. I've Fixed it

Reply:
Thanks for the update, it should fix the cases without `podNextPhase`.

but on second thought, the controller should always focus on the next phase, and it does not care about the previous pod phase, so we may have to update the test cases to adopt this.

it should be ok for this PR, and we will work on the testing later in a separate PR.

Reply:
> This test code didn't work properly if test case didn't have podNextPhase. I've Fixed it

To be precise, in the test case without `podNextPhase`, we weren't testing anything.

> but on second thought, the controller should always focus on the next phase, and it does not care about the previous pod phase, so we may have to update the test cases to adopt this.

I agree. this test has `previousPhase`, `podPhase` and `podNextPhase` fields. It might be redundant.

> it should be ok for this PR, and we will work on the testing later in a separate PR.

Sure. Thanks for your review!